spikes/bullets/Bullet.cpp

```diff
@@ -0,0 +1,37 @@
+#include "stdafx.h"
+#include "Bullet.h"
+
+#include <iostream>
+
+Bullet::Bullet()
+{
+}
+
+Bullet::Bullet(BulletType type, Vector2D initPos, Vector2D direction, float speed) : 
+m_type(type), m_position(initPos), m_velocity(direction * speed), m_dir(direction.Normalize()), m_speed(speed)
+{
+}
+
+void Bullet::Update()
+{
+	m_velocity = m_dir * m_speed;
+	m_position += m_velocity;
+}
+
+void Bullet::Draw(sf::RenderWindow *window)
+{
+	if (m_type == BulletType::CIRCLE)
+	{
+		sf::CircleShape myBullet = sf::CircleShape(5);
+		myBullet.setPosition(m_position.toSFMLVector() - Vector2D(5, 5).toSFMLVector());
```

Comment:
No, the method isnt complex it just creates an Vector2f and returns it,
thats like nothing efficiency-wise, besides we wont be using sfml shapes
anyway, so we will no longer need to use Vector2fs other than the test code
here. In other words this is not going to make the merge so its fine
On 19 Oct 2015 11:23, "ShaneCallanan" <notifications@github.com> wrote:

> In spikes/bullets/Bullet.cpp
> <https://github.com/gdxn96/swarm-wars/pull/4#discussion_r42355732>:
>
> > +m_type(type), m_position(initPos), m_velocity(direction * speed), m_dir(direction.Normalize()), m_speed(speed)
> > +{
> > +}
> > +
> > +void Bullet::Update()
> > +{
> > +	m_velocity = m_dir * m_speed;
> > +	m_position += m_velocity;
> > +}
> > +
> > +void Bullet::Draw(sf::RenderWindow *window)
> > +{
> > +	if (m_type == BulletType::CIRCLE)
> > +	{
> > +		sf::CircleShape myBullet = sf::CircleShape(5);
> > +		myBullet.setPosition(m_position.toSFMLVector() - Vector2D(5, 5).toSFMLVector());
>
> @Vector2D(5,5).toSFMLVector()
> Might be better to use Vector2f here instead, for efficiency.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gdxn96/swarm-wars/pull/4/files#r42355732>.
>